tests/test_browser.py

```diff
@@ -412,13 +412,22 @@ def make_main_two_files(path1, path2, nonexistingpath):
     self.btest_exit('main.cpp', args=['--pre-js', 'pre.js', '--use-preload-plugins'])
 
   # Tests that user .html shell files can manually download .data files created with --preload-file cmdline.
-  def test_preload_file_with_manual_data_download(self):
+  @parameterized({
+    'default': ([],),
+    'pthreads': (['-pthread', '-sPROXY_TO_PTHREAD', '-sEXIT_RUNTIME'],),
+  })
```

Comment:
Maybe needs `@requires_threads` on this test now?

Reply:
Is it required even when using `self.run_browser` ?

Reply:
Yes, that idea is that its possible to run the entire test suite in a browser that doesn't support threads, in which case such tests as these would need to be skipped.